src/main/java/Core/Minimax.java

```diff
@@ -7,70 +7,77 @@
 public class Minimax {
 
     private Grid firstGrid;
-    private Mark minimaxPlayerMark;
-    private Mark nonMinimaxPlayerMark;
+    private Mark maximisingPlayerMark;
+    private Mark minimisingPlayerMark;
 
-    public Minimax(Grid firstGrid, Mark minimaxPlayerMark, Mark nonMinimaxPlayerMark) {
+    public Minimax(Grid firstGrid, Mark maximisingPlayerMark, Mark minimisingPlayerMark) {
```

Comment:
I'd call these minimisingPlayer and maximisingPlayer because the type is Mark, so it's unnecessary to repeat in the names.